MetaMask/metamask-extension
# [FLASK] Revert changes made to stable permission display (UI design)

## Explanation

This PR is a hotfix for changes made in: https://github.com/MetaMask/metamask-extension/pull/18372

Reason: UI of the permission displayed on-connect with web3 app was non-intentionally changed because Snap install and settings flows used the same component.

Solution: Previously used component for displaying permissions when connecting is returned in its previous state. New component is created for Snaps flows only and integrated into it.

## Screenshots/Screencaps

### Before
![Screenshot 2023-04-05 at 18 14 47](https://user-images.githubusercontent.com/13301024/230141734-ffdb166d-dd28-4a7e-826b-247ee23bfff1.png)
![Screenshot 2023-04-05 at 18 15 38](https://user-images.githubusercontent.com/13301024/230141777-c6b5bdd0-e0e0-43cc-ba11-7a5b3b44a6d3.png)

### After
![Screenshot 2023-04-05 at 17 50 48](https://user-images.githubusercontent.com/13301024/230139829-9ae24a3f-31aa-4b15-a8d9-c16a6f4b8496.png)
![Screenshot 2023-04-05 at 18 01 37](https://user-images.githubusercontent.com/13301024/230139859-8267d632-5a27-487d-a471-c74365424ef8.png)

To confirm that other Snaps related pages/flows have remained the same after the hotfix:
![Screenshot 2023-04-05 at 17 53 26](https://user-images.githubusercontent.com/13301024/230140073-19968edb-f8be-4577-93e4-73a5bb4269bf.png)
![Screenshot 2023-04-06 at 13 53 00](https://user-images.githubusercontent.com/13301024/230373375-f853c08c-8a31-46f9-b729-944602940bc9.png)
![Screenshot 2023-04-06 at 15 25 52](https://user-images.githubusercontent.com/13301024/230399208-4dee8186-aaa6-4565-aa51-10022aa4c2ce.png)


## Manual Testing Steps
- Try to install a snap and check the UI related to the permissions.
- Try to update a snap and check the UI related to the permissions.
- Go to the Snap settings page and check related permission display in both, popup and expanded view.
- Check storybook component for different UI states with different values.


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
